### PR TITLE
add bash-completion for comma separated list on `kubectl get`

### PR DESCRIPTION
**Pull request by phil9909**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

`kubectl get` supports specifying a comma separated list of resource types. E.g.:

```bash
kubectl get pods,secrets
```

Will list all pods and secrets in the current namespace.

This commit adds support to the kubectl bash completion for this feature.
Which means if you type `kubectl get pods,sec` + <kbd>tab</kbd> it will be auto-completed to `kubectl get pods,secrets` (assuming the cluster does not have a CRD starting with `sec`).

**Which issue(s) this PR fixes**:

None

**Special notes for your reviewer**:

After running `make kubectl` you need to run `source <(./_output/bin/kubectl completion bash)` then you can try the new completion.

In theory other `kubectl` commands support comma separated lists of resources. One can do `kubectl edit secret,configmap test` which will open an editor with both resources (or fail if one or both don't exist).
As I think the use-cases for other commands are limited, I don't think we need completion for this.
Should this be proven wrong one can simply change the `case` block.

**Does this PR introduce a user-facing change?**:

```release-note
kubectl: add support for bash-completion for comma separated list on `kubectl get`
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/cmd.go [hunk 1]
```diff
@@ -165,6 +165,25 @@ __kubectl_get_resource()
     __kubectl_parse_get "${nouns[${#nouns[@]} -1]}"
 }
```
Comment (dougsland) [inline staging/src/k8s.io/kubectl/pkg/cmd/cmd.go:167]: Do you mind to describe your function a little bit?  It seems now we have __kubectl_get_resources() and __kubectl_get_resource() :)

File: staging/src/k8s.io/kubectl/pkg/cmd/cmd.go [hunk 2]
```diff
@@ -165,6 +165,25 @@ __kubectl_get_resource()
...
 
+__kubectl_get_resources()
+{
```
Comment (dougsland) [inline staging/src/k8s.io/kubectl/pkg/cmd/cmd.go:169]: For portability, do you mind to set a variable with name like separator=','  ? Not sure if in the future we change people change mind and use a different separator like "#", "foobar".

File: staging/src/k8s.io/kubectl/pkg/cmd/cmd.go [hunk 3]
```diff
@@ -165,6 +165,25 @@ __kubectl_get_resource()
...
+    if [[ ${#nouns[@]} -eq 0 ]]; then
+      local kubectl_out HEAD TAIL
+      if kubectl_out=$(__kubectl_debug_out "kubectl api-resources $(__kubectl_override_flags) -o name --cached --request-timeout=5s --verbs=get"); then
```
Comment (dougsland) [inline staging/src/k8s.io/kubectl/pkg/cmd/cmd.go:172]: If possible, do you mind to avoid duplicate code with kubectl_get_resource?

**Review by dougsland — COMMENTED** [review]
In general looks good the patch also tested. I have few comments.

```
./_output/local/go/bin/kubectl completion bash > ~/.kube/completion.bash.inc
./_output/local/go/bin/kubectl get pods,sec<TAB>
```

File: staging/src/k8s.io/kubectl/pkg/cmd/cmd.go (hunk 2 above)
Comment (phil9909) [reply staging/src/k8s.io/kubectl/pkg/cmd/cmd.go:167]: done

File: staging/src/k8s.io/kubectl/pkg/cmd/cmd.go (hunk 3 above)
Comment (phil9909) [reply staging/src/k8s.io/kubectl/pkg/cmd/cmd.go]: done

File: staging/src/k8s.io/kubectl/pkg/cmd/cmd.go (hunk 1 above)
Comment (phil9909) [reply staging/src/k8s.io/kubectl/pkg/cmd/cmd.go:164]: added some comments, please have a look.

**Comment by dougsland** [comment]
@phil9909 could you please squash the patches?

**Comment by phil9909** [comment]
> could you please squash the patches?

Done. While I was on it, I also rebased onto master.

**Comment by dougsland** [comment]
Looks sane to me. We need one more people to review/approve.
/lgtm

**Comment by phil9909** [comment]
----



> We need one more people to review/approve.



@brianpursley would you mind having a look?

**Review by soltysh — APPROVED** [review]
/approve

**Comment by marckhouzam** [comment]
This is a nice change!  Note that it does not seem to work for `zsh`.

But I have included this feature in the conversion from bash to Go code of #96087 which allows us to no longer worry about such differences.

**Pull request merged**